RedHatInsights/insights-inventory-frontend
# feat(ESSNTL-4194): Show groups in a table

Implements https://issues.redhat.com/browse/ESSNTL-4194.
Mock https://www.sketch.com/s/c19f555b-7887-4423-b55d-575c8dd1dfb7/a/dl8zwyb.

Create the groups list table with three columns: name, total systems and last modified. Implement sorting by each column, filtering by name and pagination.

⚠️  This doesn't implement actions (rename, delete, create group, bulk select) yet (will be in a separate PR).

## How to test

- Deploy locally with mock server (`npm run start:mock`).
- Navigate to /inventory/groups and check that the table is rendered with dynamically generated values.
- You can also disable the mock server and check whether network errors are properly handled with the error state.
- You can also tweak your mock server and generate empty response (to verify the empty state).
- The table is API paginated and filtered. Any filter change, pagination or sorting you make must trigger a new API request.

## Screenshots

![image](https://user-images.githubusercontent.com/31385370/219063592-3a4f8735-ae74-46f6-b987-c88301334e40.png)

No match state:
![image](https://user-images.githubusercontent.com/31385370/219062552-8f444ca7-7f18-4090-a51b-7e4c70c66b9b.png)

Failed request (any runtime error):
![image](https://user-images.githubusercontent.com/31385370/218801891-d8eb9673-1f1f-4740-940a-4283c2e0f8b3.png)

Pagination values:
![image](https://user-images.githubusercontent.com/31385370/218802132-eae1dce1-e3d2-4904-82ba-ba9a3a2a17bf.png)
